### DISCUSSION TITLE
create group with one out of three currently playing speakers

### DISCUSSION THREAD

**Discussion by bednim** [hklages/node-red-contrib-sonos-plus#256] [Q&A] [answered]
Hello,
I am looking to create group with one out of the three playing speakers (names: b1, b2, b3) so if b1 or b2 or b3 (most recent speaker) is playing - i want another speaker (b4) to join  (this will be executed by either button press or motion sensor)  most recently reported playing speaker and create group. 

How would I achieve this? would sonos-event help here?

Thank you

**Comment by hklages** [comment, marked as answer]
"most recent reported playing speaker" is a bit tricky. 

Idea: 

With sonos-event you can subscribe to many events - maybe playbackstate - with an intelligent filter (wait 5 seconds, ...). Playbackstate often changes for instance when switching Radio stations, playing a new track, .... 

Then you can store the name of the player in a flow variable, overwriting the value before. 

Then you use that value to group your player.

**Reply by bednim** [reply]
I almost got it but how do I store player name so I can pass it to player.join.group please? Would you be so kind to share some examples please?

**Reply by bednim** [reply]
I think I got it!
